# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/windows/msaa/GeckoCustom.cpp
# Commit: 1a948c0ec982
# Full Hash: 1a948c0ec9826d73bba860699f540cbb25f34e34
# Author: James Teh <jteh@mozilla.com>
# Date: 2021-04-29 21:42:31
# Regressor Bug: 1694865
# File Overlap Count: 1
# Description:
#   Bug 1694865 part 18: Make EnumVariant, GeckoCustom and sdn*Accessible classes aggregate MsaaAccessible/MsaaDocAccessible instead of AccessibleWrap/DocAccessibleWrap. r=morgan
#   
#   These objects need to aggregate the primary COM object (MsaaAccessible).
#   Once the IUnknown implementation moves out of AccessibleWrap, it won't be possible to aggregate *AccessibleWrap any more.
#   
# ==============================================================================

diff -r 3ea7b46bb4f8 -r 1a948c0ec982 accessible/windows/msaa/GeckoCustom.cpp
--- a/accessible/windows/msaa/GeckoCustom.cpp	Thu Apr 29 10:20:56 2021 +0000
+++ b/accessible/windows/msaa/GeckoCustom.cpp	Thu Apr 29 10:20:57 2021 +0000
@@ -4,6 +4,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+#include "AccessibleWrap.h"
 #include "GeckoCustom.h"
 
 using namespace mozilla;
@@ -11,18 +12,18 @@
 
 IMPL_IUNKNOWN_QUERY_HEAD(GeckoCustom)
 IMPL_IUNKNOWN_QUERY_IFACE(IGeckoCustom)
-IMPL_IUNKNOWN_QUERY_TAIL_AGGREGATED(mAcc)
+IMPL_IUNKNOWN_QUERY_TAIL_AGGREGATED(mMsaa)
 
 HRESULT
 GeckoCustom::get_anchorCount(long* aCount) {
-  *aCount = mAcc->AnchorCount();
+  *aCount = mMsaa->LocalAcc()->AnchorCount();
   return S_OK;
 }
 
 HRESULT
 GeckoCustom::get_boundsInCSSPixels(int32_t* aX, int32_t* aY, int32_t* aWidth,
                                    int32_t* aHeight) {
-  nsIntRect bounds = mAcc->BoundsInCSSPixels();
+  nsIntRect bounds = mMsaa->LocalAcc()->BoundsInCSSPixels();
   if (!bounds.IsEmpty()) {
     *aWidth = bounds.Width();
     *aHeight = bounds.Height();
@@ -37,7 +38,7 @@
 
 HRESULT
 GeckoCustom::get_DOMNodeID(BSTR* aID) {
-  nsIContent* content = mAcc->GetContent();
+  nsIContent* content = mMsaa->LocalAcc()->GetContent();
   if (!content) {
     return S_OK;
   }
@@ -53,18 +54,19 @@
 
 STDMETHODIMP
 GeckoCustom::get_ID(uint64_t* aID) {
-  *aID = mAcc->IsDoc() ? 0 : reinterpret_cast<uintptr_t>(mAcc.get());
+  AccessibleWrap* acc = mMsaa->LocalAcc();
+  *aID = acc->IsDoc() ? 0 : reinterpret_cast<uintptr_t>(acc);
   return S_OK;
 }
 
 STDMETHODIMP
 GeckoCustom::get_minimumIncrement(double* aIncrement) {
-  *aIncrement = mAcc->Step();
+  *aIncrement = mMsaa->LocalAcc()->Step();
   return S_OK;
 }
 
 STDMETHODIMP
 GeckoCustom::get_mozState(uint64_t* aState) {
-  *aState = mAcc->State();
+  *aState = mMsaa->LocalAcc()->State();
   return S_OK;
 }